lib/posix-process: Define kernel internal resource limit syscalls

mogasergiu opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Add the kernel internal variants of `prlimit64`, `getrlimit` and
`setrlimit`: `uk_sys_prlimit64`, `uk_sys_getrlimit` and
`uk_sys_setrlimit` respectively.
This allows kernel internal code to call these system calls' logic
without having the syscall shim wrapper logic intervene.

Note how only `uk_sys_prlimit64` has been added to `exportsyms.uk`
since the others are defined as inline.

andreittr commented on lib/posix-process/include/uk/process.h:
```diff
@@ -131,6 +132,9 @@
 
+int uk_sys_prlimit64(int pid, unsigned int resource,
+		     struct rlimit *new_limit, struct rlimit *old_limit);
+
```
More of a QoL enhancement: consider adding
```c
static inline
int uk_sys_getrlimit(int resource, struct rlimit *rlim)
{
        return uk_sys_prlimit64(0, resource, NULL, rlim);
}
```
and similar ones for `setrlimit` and `getrusage`, so you encapsulate the "conversion to prlimit64" logic in `uk_sys_*` rather than at the syscall level.
Then you can call these from their syscall implementations just like `uk_sys_prlimit64`.

andreittr reviewed (commented):
Looks good, I've only one very minor request.

mogasergiu committed:
lib/posix-process: Define kernel internal resource limit syscalls

Add the kernel internal variants of `prlimit64`, `getrlimit` and
`setrlimit`: `uk_sys_prlimit64`, `uk_sys_getrlimit` and
`uk_sys_setrlimit` respectively.
This allows kernel internal code to call these system calls' logic
without having the syscall shim wrapper logic intervene.

Note how only `uk_sys_prlimit64` has been added to `exportsyms.uk`
since the others are defined as inline.

Signed-off-by: Sergiu Moga <sergiu@unikraft.io>

andreittr reviewed (approved):
Thanks for the patch! I see you also fixed the previously-uncaught issue with the `prlimit` libc func.
I think this is good to go.

Reviewed-by: Andrei Tatar <andrei@unikraft.io>

andreittr commented:
Approved-by: Andrei Tatar <andrei@unikraft.io>
